### PR TITLE
Add wxGrid::CopySelection()

### PR DESCRIPTION
**Pull request by Blake-Madden**
The was the same code from CTRL+C. I just refactored it into a separate function that the client can call now.

### CODE REVIEW

File: src/generic/grid.cpp [hunk 1]
```diff
@@ -10852,6 +10787,77 @@ void wxGrid::ClearSelection()
...
+        if (++iter != blocks.end())
+        {
+            // As we use simple text format, we can't copy more
+            // than one block to clipboard.
+            wxLogWarning
+            (
+                _("Copying more than one selected block "
+                    "to clipboard is not supported.")
+            );
+            return;
```
Comment (vadz) [inline src/generic/grid.cpp:10827]: I think it's not very helpful for the new function to give this warning, so I'd consider changing the code further to make the new function return `bool` indicating whether anything has been copied or not and giving this message in the original event handler (alternative: make the function return the error message, as string, with empty string meaning "no error", but it looks like an overkill here).

If not, the documentation should at least mention that the function logs a warning if there is more than one selected block.

**Review by vadz — COMMENTED** [review]
Thanks, extracting this function looks like a good idea, but I'm not sure about the warning.

File: src/generic/grid.cpp [hunk 2]
```diff
@@ -6095,72 +6095,7 @@ void wxGrid::OnKeyDown( wxKeyEvent& event )
...
-                                buf += '\t';
-
-                            buf += GetCellValue(row, col);
-                        }
-
-                        buf += wxTextFile::GetEOL();
-                    }
-
-                    wxTheClipboard->SetData(new wxTextDataObject(buf));
+                    CopySelection();
```
Comment (vadz) [inline src/generic/grid.cpp:6098]: Sorry, I think there was a misunderstanding, I meant to do this:
```suggestion
                    if ( !CopySelection() )
                    {
                        wxLogWarning(...);
                    }
```

**Review by vadz — COMMENTED** [review]
Sorry, I didn't meant to remove the warning, just move it out from this function.

File: src/generic/grid.cpp [hunk 3]
```diff
@@ -10852,6 +10787,74 @@ void wxGrid::ClearSelection()
...
+            // than one block to clipboard.
+            return false;
+        }
+    }
+
+    wxClipboardLocker lockClipboard;
+    if (!lockClipboard)
+    {
+        // Error message should have been already given and we
+        // don't have much to add.
```
[the remark this answers is not in the document]
Comment (Blake-Madden) [reply src/generic/grid.cpp]: I removed this and make the CTRL+C log message more encompassing (but still informative).

**Comment by vadz** [comment]
Thanks, looks good to me, will merge in a moment.

**Comment by vadz** [comment]
Sorry, after further thought I've realized that I am not sure with the latest commit added here. While I agree that it's useless when copying just a single cell, I think the trailing newline should be preserved when copying multiple cells.

So for now I've (squash) merged the rest, but please resubmit the latest commit as another PR preserving the multiple cells case unless you have a convincing reason to remove it in that case too.

TIA!

**Pull request closed without merging**